packages/ui/src/components/UnitInput/index.tsx

```diff
@@ -0,0 +1,346 @@
+    & > ${StyledNumberInputWrapper} {
+      border-right-color: ${({ theme }) => theme.colors.danger.border};
+    }
+    & > ${StyledNumberInputWrapper}:hover {
+      border-right-color: ${({ theme }) => theme.colors.danger.border};
+    }
+  }
+  &[data-error='true'] {
+    border: 1px solid ${({ theme }) => theme.colors.danger.border};
+    & > ${StyledNumberInputWrapper} {
+      border-right-color: ${({ theme }) => theme.colors.danger.border};
+    }
+  }
+`
+
+const CustomSelectInput = styled(SelectInputV2)<{
+  width?: number
+  'data-disabled': boolean
+}>`
+  #unit {
+    border: none;
+    background: transparent;
+  }
+  ${({ width }) => width && `width: ${width}px;`}
+
+  &:not([data-disabled="true"]):focus, 
+  :active {
+    box-shadow: ${({ theme }) => theme.shadows.focusPrimary};
+    & > ${StyledNumberInputWrapper} {
+      border-right-color: ${({ theme }) => theme.colors.primary.border};
+    }
+  }
+
+  &:focus,
+  :active {
+    & > ${UnitInputWrapper} {
+      box-shadow: ${({ theme }) => theme.shadows.focusPrimary};
+      border-color: ${({ theme }) => theme.colors.primary.borderHover};
+    }
+  }
+`
+const StyledInput = styled.input`
+  flex: 1;
+  border: none;
+  outline: none;
+  height: 100%;
+  padding-left: ${({ theme }) => theme.space['2']};
+  background: transparent;
+  color: ${({ theme }) => theme.colors.neutral.text};
+  &::placeholder {
+    color: ${({ theme }) => theme.colors.neutral.textWeak};
+  }
+
+  &:focus,
+  :active {
+    & > ${UnitInputWrapper} {
+    box-shadow: ${({ theme }) => theme.shadows.focusPrimary};
+    border-color: ${({ theme }) => theme.colors.primary.borderHover};
+  }
+
+  &:disabled {
+    cursor: not-allowed;
+    color: ${({ theme }) => theme.colors.neutral.textWeakDisabled};
+  }
+`
+type UnitInputValue = { inputValue: number; unit: string }
+
+type UnitInputProps = {
+  className?: string
+  max?: number
+  min?: number
+  value?: UnitInputValue['inputValue']
+  unitValue?: UnitInputValue['unit']
+  onChange?: (value: UnitInputValue['inputValue']) => void
+  onChangeUnitValue?: (values: string) => void
+  options: OptionType[]
+  selectInputWidth?: number
+  size?: 'small' | 'medium' | 'large'
+  'data-testid'?: string
+  helper?: string
+  unitError?: string
+  width?: ComponentProps<typeof Stack>['width']
+  placeholderUnit?: string
+  error?: boolean | string
+  success?: boolean | string
+  label?: string
+  labelInformation?: ReactNode
+  step?: number | string
+} & Pick<
+  InputHTMLAttributes<HTMLInputElement>,
+  | 'onFocus'
+  | 'onBlur'
+  | 'name'
+  | 'id'
+  | 'placeholder'
+  | 'disabled'
+  | 'readOnly'
+  | 'required'
+  | 'autoFocus'
+  | 'onKeyDown'
+>
+
+export const UnitInput = ({
+  id,
+  name = '',
+  max = 99999,
+  min = 1,
+  autoFocus = false,
+  size = 'large',
+  placeholder = '0',
+  placeholderUnit = 'Select item',
+  onChange,
+  onChangeUnitValue,
+  value,
+  unitValue,
+  selectInputWidth = 200,
+  disabled = false,
```

Comment:
There is an issue with disabled, seems like mouse pointer is not disabled when hovering number input part but it is when hovering select input part